### PR TITLE
keyd: init at 2.2.5-beta

### LINKED ISSUE #132474: package request: keyd - "A key remapping daemon for linux"
**Project description**

https://github.com/rvaiya/keyd

> A key remapping daemon for linux.

> Linux lacks a good key remapping solution. In order to achieve satisfactory results a medley of tools need to be employed (e.g xcape, xmodmap) with the end result often being tethered to a specified environment (X11). keyd attempts to solve this problem by providing a flexible system wide daemon which remaps keys using kernel level input primitives (evdev, uinput).

> Features
> keyd has several unique features many of which are traditionally only found in custom keyboard firmware like QMK. Some of the more interesting ones include:

>Layers.
>Key overloading (different behaviour on tap/hold).
Per keyboard configuration.
Instantaneous remapping (no flashing required).
A simple and intuitive config format.
Being display server agnostic (works on wayland and virtual console alike).

**Metadata**

* homepage URL: https://github.com/rvaiya/keyd
* source URL: https://github.com/rvaiya/keyd
* license: MIT
* platforms: linux

### PR DESCRIPTION
**Pull request by dit7ya**
###### Motivation for this change

Adding the `keyd` key remapping daemon. It functions like `kmonad` and similar tools for creating highly customizable keyboard configurations.

Closes  #132474.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/inputmethods/keyd/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchFromGitHub {
+    owner = "rvaiya";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-Mb5LU8xsX8H/k3yMQiaCw4BVMphxNo25HPE0sMWTD8Y=";
+  };
+  installPhase = ''
+    make install DESTDIR=$out PREFIX=
+  '';
+
```
Comment (legendofmiracles) [inline pkgs/tools/inputmethods/keyd/default.nix:17]:
```suggestion
  makeFlags = [ "DESTDIR=${placeholder "out"}" "PREFIX=" ];
```
This should also do it, haven't tested it but that's the idea

File: pkgs/tools/inputmethods/keyd/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  meta = with lib; {
+
```
Comment (legendofmiracles) [inline pkgs/tools/inputmethods/keyd/default.nix:19]:
```suggestion
```

File: pkgs/tools/inputmethods/keyd/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{ lib, fetchFromGitHub, stdenv }:
+
+stdenv.mkDerivation rec {
+
```
Comment (legendofmiracles) [inline pkgs/tools/inputmethods/keyd/default.nix:4]:
```suggestion
```

File: pkgs/tools/inputmethods/keyd/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      It expects a configuration file at /etc/keyd/default.conf. For
+      more details check the homepage.
+    '';
+    license = licenses.mit;
+    maintainers = with maintainers; [ dit7ya ];
+    platforms = platforms.linux;
+  };
+
+}
+
```
Comment (legendofmiracles) [inline pkgs/tools/inputmethods/keyd/default.nix:36]:
```suggestion
}
```

File: pkgs/tools/inputmethods/keyd/default.nix (hunk 1 above)
Comment (peterhoeg) [reply pkgs/tools/inputmethods/keyd/default.nix]: I think it's the other way around (set prefix, unset deskdir)

**Pull request closed without merging**